docs/docs-content/clusters/edge/site-deployment/deploy-private-registry.md

```diff
@@ -0,0 +1,96 @@
+
+## Limitations
+
+- A cluster cannot pull images from more than one private registry.
+
+- If your private registry has TLS enabled, you can only configure a new cluster to use a TLS certificate with a private registry. You cannot configure an existing cluster with a TLS certificate to communicate with your private registry. 
+
+- Palette Edge supports basic username/password authentication. Token authentication schemes used by services such as AWS ECR and Google Artifact Registry are not supported. 
+
+## Prerequisites
+
+- At least one Edge host with an x86_64 or AMD64 processor architecture. 
+
+- A private image registry.
+
+- A provider image you created in the EdgeForge process stored in your private image registry. For more information, refer to [Build Artifacts](../edgeforge-workflow/palette-canvos.md).
+
+## Enablement
+
+1. Log in to [Palette](https://console.spectrocloud.com).
+
+2. Navigate to the left **Main Menu** and select **Profiles**.
+
+3. If you already have a cluster profile you want to deploy the cluster with, select that profile and select **Create new version** to create a new version of the profile to save your changes.
+
+   Otherwise, click **Add new profile** to create a new cluster profile. 
+
+4. Select the OS layer of your cluster profile. If you are creating a new profile, you will get to configuring the OS layer after filling out **Basic Information** and **Cloud Type**. 
+
+5. Update the `system.uri` parameter in the pack editor for your OS layer. Use the custom OS image you created in the EdgeForge process. Refer to the EdgeForge [Build Images](../edgeforge-workflow/palette-canvos.md) guide if you are missing a custom OS image. The following is an example configuration using a custom OS image.
+
+  ```yaml
+  pack:
+  content:
+    images: 
+      - image: '{{.spectro.pack.edge-native-byoi.options.system.uri}}'
+      # - image: example.io/my-other-images/example:v1.0.0 
+      # - image: example.io/my-super-other-images/example:v1.0.0
+      
+  options: 
+    system.uri: example.io/my-images/example-custom-os:v1.4.5
+  ```
+
+6. At the root level of YAML for your OS layer, add the `providerCredentials` field to provide the credentials you need to authenticate with your registry. The `providerCredentials.password` field will be masked when you provide it in the YAML file:
+
+  ```yaml {7-16}
+  pack:
+  content:
+    images: 
+      - image: '{{.spectro.pack.edge-native-byoi.options.system.uri}}'
+      # - image: example.io/my-other-images/example:v1.0.0 
+      # - image: example.io/my-super-other-images/example:v1.0.0
+  providerCredentials:
+    registry: domain/project
+    user: user
+    password: ******
+    certificates: |
+      -----BEGIN CERTIFICATE-----
+      MIIDVzCCAj+gAwIBAgIRANtGPo/hFkZtYRNw0KaeW54wDQYJKoZIhvcNAQELBQAw
+      ----------------------------------------------------------------
+      7OicCaV35lje5FSl0owu74ghAlCgMyAdKsJf615g1kKO4V5E2BMErd9Ibw==
+      -----END CERTIFICATE-----
```

Comment:
Hmmm...we should think about defining the parameters, earlier in the doc. I'm not sure they are the same as exposed in the user-data. For example, in the user-data `registryCredentials` parameter block, the `insecure` option is available for endpoints using self-signed certs or endpoints using certs signed by an unknown certificate authority. 

It's also possible the `registryCredentials` got new parameters. Actually, looking the code base, it looks like `repositoryName` is a new attribute. 

https://github.com/spectrocloud/stylus/blob/main/domain/cloudinit_types.go#L67